src/main/java/de/dennisguse/opentracks/stats/TrackStatisticsUpdater.java

```diff
@@ -148,23 +148,17 @@ public void addTrackPoint(TrackPoint trackPoint, Distance minGPSDistance) {
         if (trackPoint.hasSensorDistance()) {
             // Sensor-based distance/speed
             currentSegment.addTotalDistance(trackPoint.getSensorDistance());
-        } else if (lastTrackPoint != null && trackPoint.isMoving()) {
+        } else if (lastTrackPoint != null && lastTrackPoint.hasLocation() && trackPoint.isMoving()) {
```

Comment:
We might still have some issue here with mixing SENSORPOINT and TRACKPOINT as distanceTo does not work if between GPS-only TRACKPOINT and SENSORPOINT.
Just a gut feeling.

Reply:
Or do we just ignore the distance in this case?

Reply:
BTW (Offtopic): `minGPSDistance` argument in this method is unused.